### PR TITLE
first draft of proposal process

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

**Comment by strk** [comment]
"Gitea 1.x lower version must continue to work after upgrade to 1.x higher version. Any proposed change must not break this promise."

The above statement is confusing. How about upgrade to 2.x ?
It looks like there's no provision for every getting to 2.x ?

**Comment by tboerger** [comment]
LGTM

**Comment by tboerger** [comment]
> The above statement is confusing. How about upgrade to 2.x ?
> It looks like there's no provision for every getting to 2.x ?

Since that should be a major release breaking changes are allowed, but we need to provide some upgrade path with potentially manual tasks.

File: README.md [hunk 1]
```diff
@@ -1 +1,116 @@
...
+- Mail [gitea-develop](https://groups.google.com/d/forum/gitea-develop) as usual.
+- Discussion [gitter for gitea-develop](https://gitter.im/go-gitea/develop)
+
+Worst case, by bypassing the initial discussion you've possibly written an
+unnecessary design doc. Not a big deal.
+
+## Help
+
+If you need help with this process, please contact the Gitea contributors by posting
+to the [golang-dev mailing list](https://groups.google.com/d/forum/gitea-develop).
```
Comment (xinity) [inline README.md]: this line should be changed to gitter rooms

**Comment by LefsFlarey** [comment]
Would be nice if everything is properly word/character limited.

File: README.md [hunk 2]
```diff
@@ -1 +1,116 @@
...
+### Goals
+
+- Make sure that proposals get a proper, fair, timely, recorded evaluation with
+  a clear answer.
+- Make past proposals easy to find, to avoid duplicated effort.
+- If a design doc is needed, make sure contributors know how to write a good one.
+
+### Definitions
+
+- A **proposal** is a suggestion filed as a GitHub issue, identified by having
```
Comment (andreynering) [inline README.md]: An issue on go-gitea/gitea or go-gitea/proposals?

Comment (lunny) [inline README.md]: go-gitea/gitea, I have disabled go-gitea/proposals' issues on github settings. I will indicate it exactly.

**Comment by strk** [comment]
Where is this going to end up ? Are we keeping or leaving go-gitea/proposals ?

File: README.md (hunk 2 above)
Comment (lunny) [inline README.md]: Done.

**Comment by strk** [comment]
LGTM

**Comment by lunny** [comment]
It seems lgtm doesn't work on this.

**Pull request merged**